core/src/main/java/io/grpc/internal/HttpUtil.java

```diff
@@ -44,6 +44,18 @@
 public final class HttpUtil {
 
   /**
+   * {@link io.grpc.Metadata.Key} for the :authority pseudo header.
+   */
+  public static final Metadata.Key<String> AUTHORITY_KEY =
+          Metadata.Key.of(":authority", Metadata.ASCII_STRING_MARSHALLER);
```

Comment:
Hmm... so colon in the name should probably be prevented. I guess this shows we need an issue similar to #725 but for metadata keys. Created #824